verify.go

```diff
@@ -117,10 +121,16 @@ func verifyCert(certBytes []byte, signingCert *api.GovalCert) (*api.GovalCert, e
 		}
 	}
 
+	// Store this cert's claims so we can validate tokens later.
+	certClaims := parseClaims(&cert)
+	if certClaims != nil {
+		v.currentClaims = certClaims
+	}
```

Comment:
is there any way in which we can have this be in the "constructor" of the verifier? otherwise it's hard to know what state the object is (it's kinda like the `.Init()` anti-pattern where an object is created in an invalid state and then needs another step to make it valid).

IIUC making `verifyChain` become `newVerifier` _should_ be sufficient (and adding getters for the payload + cert).

Reply:
hmm, the `claims` member is OK to be `nil`. As we navigate the chain, we load the most-recently-seen claims (if present) so that final verification runs using the claims in the final `api.GovalCert` in the chain. So the verifier object should already be in a valid state in all cases. Trying to understand what change needs to be made here, that's all.

Reply:
yup, `claims` can be `nil` if the token was signed directly with a root cert. but we don't _enforce_ that it's in the state we want (be it being `nil` or non-`nil`).

the anti-pattern i'm talking about is that there is an implicit requirement to call `verifyChain` before `parseClaims` to make it be in the correct state. if that's the case, `verifyChain` should be part of the constructor. that prevents the object from being in an invalid state ever!

(okay, it's a private struct, so one can create a zero-value `verifier`, but since it has a constructor, folks _should_ always use the constructor).

Reply:
tweaked, there's now a `verifyChain()` entry point that creates the `verifier` so it doesn't need to be done in other code paths